File: algorithms/0881. Boats to Save People.cpp
class Solution
{
public:
    int numRescueBoats(vector<int>& people, int limit)
    {
        sort(people.begin(), people.end());
        
        int ans = 0;
        int l = 0;
        int r = people.size() - 1;
        
        while (r >= l)
        {
            if (people[r] + people[l] <= limit)
            {
                ++l;
            }
            
            --r;
            ++ans;
        }
        
        return ans;
    }
};

File: 0881. Boats to Save People.cs
public class Solution
{
    public int NumRescueBoats(int[] people, int limit)
    {
        Array.Sort(people);

        var ans = 0;
        var l = 0;
        var r = people.Length - 1;

        while (r >= l)
        {
            if (people[r] + people[l] <= limit)
            {
                ++l;
            }

            ++ans;
            --r;
        }

        return ans;
    }
}